ext/mbstring/tests/gh10192.phpt

```diff
@@ -0,0 +1,29 @@
+--TEST--
+GH-10192 (mb_detect_encoding() results for UTF-7 differ between PHP 8.0 and 8.1)
+--EXTENSIONS--
+mbstring
+--FILE--
+<?php
```

Comment:
The test cases should more thoroughly explore the behavior of `mb_detect_encoding` and `mb_check_encoding` as re: UTF-7.

You should also include test cases for UTF7-IMAP as well.

Reply:
The test cases should also demonstrate the behavior of `mb_convert_encoding`, showing that while conversion does not insert '?' and does not increment `mb_get_info('illegal_chars')`, `mb_check_encoding` still returns `false` for UTF-7 and UTF7-IMAP strings with improperly terminated Base64 sections.